### ISSUE TITLE
Rendering scopes with `includes` is slow for no reason

### ISSUE THREAD

**Issue by giovannibonetti** [tiagopog/jsonapi-utils#78]
We have in our codebase something like this:
```ruby
class AppointmentsController < BaseController
  # GET /api/v2/appointments
  def index
    appointments = Appointment.all.includes(:professional, :service)
    jsonapi_render json: appointments
  end
end
````

We use `includes` to eager load associations and avoid the N+1 query problem.
However, the _record_count_ _query_ becomes very slow since there is now a `LEFT OUTER JOIN` for each eager loded association:

```ruby
SELECT DISTINCT COUNT(DISTINCT appointments.id) FROM "appointments" LEFT OUTER JOIN "professionals" ON "professionals"."id" = "appointments"."employment_id" LEFT OUTER JOIN "services" ON "services"."id" = "appointments"."service_id"
```

I suppose a simple fix could be done by changing [this line](https://github.com/tiagopog/jsonapi-utils/blob/master/lib/jsonapi/utils/support/pagination.rb#L139) from this:

```ruby
def count_records(records, options)
  ...
  records.except(:group, :order).count("DISTINCT #{records.table.name}.id")
end
```

to this:
```ruby
def count_records(records, options)
  ...
  records.except(:group, :includes, :order).count("DISTINCT #{records.table.name}.id")
end
```

By the way, can you please backport this patch to the `0.4` series as well?

**Comment by tiagopog** [comment]
Sure, I will work on a patch version bringing this improvement for both series `0.5` and `0.4`.  Thanks for reporting 👍

**Comment by tiagopog** [comment]
The update that fixes this issue was first packed in `v0.7.1` but I'm still going to release it for `v0.4.x`.
